src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -498,16 +533,19 @@ Transactor::checkSingleSign(PreclaimContext const& ctx)
 {
     // Check that the value in the signing key slot is a public key.
     auto const pkSigner = ctx.tx.getSigningPubKey();
-    if (!publicKeyType(makeSlice(pkSigner)))
+    if (!(ctx.flags & tapDRY_RUN) && !publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace())
             << "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH;  // FIXME: should be better error!
     }
 
     // Look up the account.
-    auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
     auto const idAccount = ctx.tx.getAccountID(sfAccount);
+    // This ternary is only needed to handle `simulate`
+    auto const idSigner = pkSigner.size() > 0
```

Comment:
Consider to use `pkSigner.empty()` as `.size()` is not a constant time function for all the std containers

Reply:
fixed in 316639642e6a4d8111a5da29af35c965dbc11d01